sympy/combinatorics/isomorphisms.py

```diff
@@ -0,0 +1,90 @@
+import itertools
+from sympy.combinatorics.homomorphisms import homomorphism, GroupHomomorphism
+from sympy.combinatorics.fp_groups import FpGroup
+from sympy.combinatorics.perm_groups import PermutationGroup
+from sympy import S
+
+def group_isomorphism(G, H):
+    '''
+    Compute the isomorphism between 2 given groups.
+
+    Arguments:
+        G (a finite `FpGroup` or a `PermutationGroup`) -- First group
+        H (a finite `FpGroup` or a `PermutationGroup`) -- Second group
+
+    Returns:
+    If the groups are not isomorphic, `None` is returned.
+    If the groups are isomorphic, the corresponding `GroupHomomorphism` object is returned.
+
+    Summary:
+    Uses the approach suggested by Robert Tarjan to compute the isomorphism between two groups.
+    First, the set of generators are mapped with the elements of `H` and we check if the mapping induces isomorphism.
+
+    '''
+
+    if isinstance(G, FpGroup):
+        if G.order() == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        G = G._to_perm_group()[0]
+    if isinstance(H, FpGroup):
+        if H.order() == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        H = H._to_perm_group()[0]
+
+    if not isinstance(G, PermutationGroup):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+    if not isinstance(H, PermutationGroup):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
```

Comment:
I think it make sense to put these two checks at the very beginning.

Reply:
But, if the group is an `FpGroup`, we should check if the group is finite and convert it to the `PermutationGroup` isomorphic to it, right?

Reply:
Ah, I see what you are doing. I initially thought that you are checking if `G` and `H` are proper input types, but yes, turning `FpGroup` ones into permutation groups and then checking the types does make sense. Sorry about that.